app.rb

```diff
@@ -1,5 +1,16 @@
 require_relative 'config/environment'
+require 'sinatra/base'
 
 class App < Sinatra::Base
+  get '/' do
+    erb :index
+  end
 
+  get '/new' do
+    erb :new
+  end
+
+  post '/display_puppy' do
+    erb :display_puppy
+  end
```

Comment:
It would be better to use the `Puppy` class here. You could initialize it with the params, then have access it in the view.